### ISSUE TITLE
Tags suggestions are not accessible

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1107]
Just noticed the Categories & Tags auto-suggestions (If I'm not wrong, at the moment it's implemented just for tags).

Initially, it's rendered as an `<input>` element:

![screen shot 2017-06-10 at 14 04 27](https://user-images.githubusercontent.com/1682452/27002715-85f00290-4de8-11e7-9bac-901764dd2857.png)

when tags get added, some elements get pre-pended to the input field:

![screen shot 2017-06-10 at 14 04 50](https://user-images.githubusercontent.com/1682452/27002721-a72a0802-4de8-11e7-8dd1-ed3445ffc0b3.png)

Apart from semantics considerations (the widget would need to use some ARIA), the list of suggestions is not accessible and there's nothing there screen readers can announce.

Core introduced a standard for this kind of "autocomplete suggestions" and they're currently used:
- in TinyMCE when searching for a link in the link inline toolbar
- for the tags in the Quick Edit form and the tags in the Edit post tags metabox

a very nice job has been done to make these auto-suggestions widgets accessible, using all the recommended ARIA authoring practices. It would be great to see Gutenberg follow the standards established by core and not introduce a decreased level of accessibility compared to what core provides right now.

The core implementation uses an ARIA `combobox` widget, that's a composite widget made of an input field and an ARIA listbox. The core implementation is pretty straightforward and can be seen in:
`wp-admin/js/tags-suggest.js`
`wp-includes/js/tinymce/plugins/wplink/plugin.js`

A separate issue are the current "tag labels":

<img width="287" alt="screen shot 2017-06-10 at 14 41 52" src="https://user-images.githubusercontent.com/1682452/27002818-069167b6-4deb-11e7-97b2-6b06bbbba1c0.png">

the current implementation should be carefully tested and improved, because right now they're just `<span>` elements with an empty button to remove them.

Some screenshots to illustrate how the current tags autocomplete suggestions works in core:

![screen shot 2017-06-10 at 14 33 07](https://user-images.githubusercontent.com/1682452/27002874-5ff6f5d6-4dec-11e7-9166-63be315fea93.png)

![screen shot 2017-06-10 at 14 33 20](https://user-images.githubusercontent.com/1682452/27002875-5ffe7518-4dec-11e7-874d-219aeacfbd4d.png)

![screen shot 2017-06-10 at 14 33 29](https://user-images.githubusercontent.com/1682452/27002876-5ffede18-4dec-11e7-875b-0310ac152e3e.png)

![screen shot 2017-06-10 at 14 54 59](https://user-images.githubusercontent.com/1682452/27002915-0df3cad8-4ded-11e7-948d-aa2639271a69.png)

![screen shot 2017-06-10 at 14 33 35](https://user-images.githubusercontent.com/1682452/27002877-60161506-4dec-11e7-8464-0f86733c3090.png)

![screen shot 2017-06-10 at 14 55 12](https://user-images.githubusercontent.com/1682452/27002919-16d0390c-4ded-11e7-937c-181abd29ce29.png)

![screen shot 2017-06-10 at 14 55 18](https://user-images.githubusercontent.com/1682452/27002918-16cdfd4a-4ded-11e7-8a12-784211acbc2e.png)
